core/roboconf-dm-templating/src/main/java/net/roboconf/dm/templating/internal/TemplatingManager.java

```diff
@@ -26,595 +26,347 @@
 package net.roboconf.dm.templating.internal;
 
 import java.io.File;
-import java.io.FileFilter;
 import java.io.IOException;
-import java.io.InputStream;
 import java.lang.reflect.UndeclaredThrowableException;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.Objects;
-import java.util.Set;
-import java.util.concurrent.locks.ReadWriteLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
 import java.util.logging.Logger;
 
 import net.roboconf.core.model.beans.Application;
+import net.roboconf.core.model.beans.ApplicationTemplate;
+import net.roboconf.core.model.beans.Instance;
 import net.roboconf.core.utils.Utils;
-import net.roboconf.dm.management.TemplatingManagerService;
-import net.roboconf.dm.templating.TemplatingService;
-
-import org.apache.commons.io.filefilter.CanReadFileFilter;
-import org.apache.commons.io.filefilter.FileFilterUtils;
-import org.apache.felix.ipojo.annotations.Component;
-import org.apache.felix.ipojo.annotations.Instantiate;
-import org.apache.felix.ipojo.annotations.Invalidate;
-import org.apache.felix.ipojo.annotations.Property;
-import org.apache.felix.ipojo.annotations.Provides;
-import org.apache.felix.ipojo.annotations.Validate;
-
-import com.github.jknack.handlebars.Context;
-import com.github.jknack.handlebars.HandlebarsException;
-
+import net.roboconf.dm.management.ManagedApplication;
+import net.roboconf.dm.management.Manager;
+import net.roboconf.dm.management.events.EventType;
+import net.roboconf.dm.management.events.IDmListener;
+import net.roboconf.dm.templating.internal.templates.TemplateEntry;
+import net.roboconf.dm.templating.internal.templates.TemplateUtils;
+import net.roboconf.dm.templating.internal.templates.TemplateWatcher;
 
 /**
- * Component dedicated to the generation of reports for Roboconf applications.
  * @author Pierre Bourret - Université Joseph Fourier
+ * @author Vincent Zurczak - Linagora
  */
-@Component(name = "roboconf-templating-manager", managedservice = "net.roboconf.dm.templating.configuration", publicFactory = false)
-@Provides(specifications = {TemplatingManagerService.class, TemplatingService.class})
-@Instantiate(name = "Roboconf Templating Manager")
-public class TemplatingManager implements TemplatingManagerService, TemplatingService {
+public class TemplatingManager implements IDmListener {
 
+	public static final String ID = "DM's Templating";
 
-	/**
-	 * The big bad global lock.
-	 */
-	private final ReadWriteLock lock = new ReentrantReadWriteLock(true);
+	// Locks
+	private final Object watcherLock = new Object();
```

Comment:
`ReadWriteLock`s allow better throughput, especially when reads vastly outnumber writes (which is the case here).
`synchronized` excludes two concurrent reads :(